src/detect-engine.h

```diff
@@ -27,6 +27,24 @@
 #include "detect.h"
 #include "suricata.h"
 
+#include "detect-content.h"
+
+typedef struct DetectBufferType_ {
+    char name[32];
+    char description[128];
+    int id;
+    int parent_id;
+    bool mpm;
+    bool packet; /**< compat to packet matches */
+    bool frame;  /**< is about Frame inspection */
+    bool supports_transforms;
+    bool multi_instance; /**< buffer supports multiple buffer instances per tx */
+    void (*SetupCallback)(const struct DetectEngineCtx_ *, struct Signature_ *);
+    bool (*ValidateCallback)(
```

Comment:
validation should be possible for non-content as well

Reply:
Interesting, what can we validate besides content ?

Reply:
all other keywords :)

examples would be bsize, byte*, etc.

Reply:
Ok, so we could have `DetectMd5ValidateCallback` forbid the usage of other stuff than `content` for instance... Interesting

(As of today, we only use content though, right ?)